Skip blobs before fetching them

esoergel opened the pull request:
##### SUMMARY
PRing into https://github.com/dimagi/commcare-hq/pull/28725

This move the existence check before the `.get`, which seems like a huge performance win when topping up blobs.  Am I missing something or is this correct?

##### FEATURE FLAG
<!-- If this is specific to a feature flag, which one? -->

##### RISK ASSESSMENT / QA PLAN
<!--
    Is there test coverage? Is there manual QA planned? Link QA ticket.
    Is there anything a deployer should know about rolling back this change?
-->

##### PRODUCT DESCRIPTION
<!-- For non-invisible changes, describe user-facing effects. -->

esoergel committed:
Skip blobs before fetching them

kaapstorm reviewed (approved):
Whoa! Yeah, I think you're right!

esoergel committed:
Pull already_exported to where it's used

I think this makes sense, since it's not really an attribute of the
archive anyways.
